src/core/server/plugins/plugin.test.ts

```diff
@@ -263,3 +262,65 @@ test('`stop` calls `stop` defined by the plugin instance', async () => {
   await expect(plugin.stop()).resolves.toBeUndefined();
   expect(mockPluginInstance.stop).toHaveBeenCalledTimes(1);
 });
+
+describe('#getConfigSchema()', () => {
+  it('reads config schema from plugin', () => {
+    const pluginSchema = schema.any();
+    jest.doMock(
+      'plugin-with-schema/server',
+      () => ({
+        config: {
+          schema: pluginSchema,
+        },
+      }),
+      { virtual: true }
+    );
+    const manifest = createPluginManifest();
+    const plugin = new PluginWrapper(
+      'plugin-with-schema',
+      manifest,
+      createPluginInitializerContext(coreContext, manifest)
+    );
+
+    expect(plugin.getConfigSchema()).toBe(pluginSchema);
+  });
+  it('returns null if config definition not specified', () => {
+    jest.doMock('plugin-with-no-definition/server', () => ({}), { virtual: true });
+    const manifest = createPluginManifest();
+    const plugin = new PluginWrapper(
+      'plugin-with-no-definition',
+      manifest,
+      createPluginInitializerContext(coreContext, manifest)
+    );
+    expect(plugin.getConfigSchema()).toBe(null);
+  });
+  it('returns null for plugins without a server part', () => {
+    const manifest = createPluginManifest({ server: false });
+    const plugin = new PluginWrapper(
+      'plugin-with-no-definition',
+      manifest,
+      createPluginInitializerContext(coreContext, manifest)
+    );
+    expect(plugin.getConfigSchema()).toBe(null);
+  });
+  it('throws if plugin contains invalid schema', () => {
+    jest.doMock(
+      'plugin-invalid-schema/server',
+      () => ({
+        config: {
+          schema: () => null,
```

Comment:
__optional suggestion:__ maybe we can use `{ validate() {} }` instead of `null`, to make sure we don't accept duck typing.